### PR TITLE
Migrate ElasticSearch routes and state machine to Cats Effect

### LINKED ISSUE #4408: Move Elasticsearch state machine and routes to Cats effect
Not in scope: Elasticsearch indexing process

### PR DESCRIPTION
**Pull request by dantb**
Fixes #4408

### CODE REVIEW

File: delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/model/package.scala [hunk 1]
```diff
@@ -52,37 +52,34 @@ package object model {
 
   implicit private val logger: Logger = Logger("ElasticSearchPlugin")
 
+  // TODO can do this only once and inject them where they're need instead of memoizing?
```
Comment (dantb) [inline delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/model/package.scala:55]: Got another PR incoming to do this TODO, thought it was worth splitting out

**Review by olivergrabinski — CHANGES_REQUESTED** [review]
Looks OK, though I think you need to `attemptNarrow` on some endpoints like `_mapping` because it leads to a test failure otherwise

**Comment by dantb** [comment]
> Looks OK, though I think you need to `attemptNarrow` on some endpoints like `_mapping` because it leads to a test failure otherwise

Thanks, also had forgotten the query routes - fixed in https://github.com/BlueBrain/nexus/pull/4449/commits/5e5e464c4cbfa6394962b80d47bdd65ae9899df0

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/Resources.scala [hunk 2]
```diff
@@ -264,8 +264,11 @@ object Resources {
     * Expands the segment to a [[ResourceRef]]
     */
   def expandResourceRef(segment: IdSegment, context: ProjectContext): IO[ResourceRef] =
+    expandResourceRef(segment, context, InvalidResourceId)
+
+  def expandResourceRef(segment: IdSegment, context: ProjectContext, notFound: String => Throwable): IO[ResourceRef] =
     IO.fromOption(segment.toIri(context.apiMappings, context.base).map(ResourceRef(_)))(
-      InvalidResourceId(segment.asString)
+      notFound(segment.asString)
     )
```
Comment (olivergrabinski) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/Resources.scala:272]: Ok for me but why the second method here?

**Review by olivergrabinski — APPROVED** [review]

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/Resources.scala (hunk 2 above)
Comment (dantb) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/Resources.scala:272]: Ah yeah, was going to reuse it in the ES module but forgot to call it - will open another PR as I think it's a bit bigger

**Pull request merged**